godo.go

```diff
@@ -315,6 +332,30 @@ func SetStaticRateLimit(rps float64) ClientOpt {
 	}
 }
 
+// SetRetryMax sets an optional client-side....
+func SetRetryMax(retryMax int) ClientOpt {
```

Comment:
What would it look like to use a single option configured with a struct? Something like:

```go
type RetryConfig struct {
	RetryMax     int
	RetryWaitMin float64 // Minimum time to wait
	RetryWaitMax float64 // Maximum time to wait
}

func WithRetryAndBackoffs(retryConfig RetryConfig) ClientOpt {
// ...
```